runtime/compiler/ilgen/Walker.cpp

```diff
@@ -5894,7 +5894,8 @@ TR_J9ByteCodeIlGenerator::loadArrayElement(TR::DataType dataType, TR::ILOpCodes
    if (mayBeValueType &&
        TR::Compiler->om.isValueTypeArrayFlatteningEnabled() && // isValueTypeArrayFlatteningEnabled() checks areFlattenableValueTypesEnabled()
        !TR::Compiler->om.canGenerateArraylets() &&
-       dataType == TR::Address)
+       dataType == TR::Address &&
+       !_methodSymbol->skipFlattenableArrayElementNonHelperCall())
```

Comment:
Updated this line in e2a91ea. I realized `skipFlattenableArrayElementNonHelperCall` should be called instead of `safeToSkipNonNullableArrayNullStoreCheck`